test/utils/oidc/handlers.go

```diff
@@ -0,0 +1,23 @@
+//go:generate mockgen -source=handlers.go -destination=handlers.mock.go -package=oidc TokenHandler JWKsHandler
+
+package oidc
+
+import (
+	"gopkg.in/square/go-jose.v2"
+)
+
+type Token struct {
```

Comment:
do we need this exported?

Reply:
Yes, it's expoted here: [`test/integration/oidc/oidc_test.go:416`](https://github.com/kubernetes/kubernetes/pull/115122/commits/acebeb0747014efa2b2c00a0df34d07e28d2f3a2#diff-b67c558e6af583e63cf40e6fb4e06588fda7ae8ae14325d6711496982e51e41cR416)

Reply:
is this going to be used in any other place that in that test?

if not we can move it there and avoid this cross dependencies

Reply:
> is this going to be used in any other place that in that test?

Yes, the idea is to create pretty universal test OIDC server that is able to be used wherever OIDC involved if needed. BTW we've [agreed](https://github.com/kubernetes/kubernetes/issues/110782#issuecomment-1386537838) with @enj to replace less flexible test OIDC server:
https://github.com/kubernetes/kubernetes/blob/106332252bdfa3a4e4ce7e01a3f389a0bb99d06a/staging/src/k8s.io/apiserver/plugin/pkg/authenticator/token/oidc/oidc_test.go#L155-L159
so here is potentially new place where `oidc.Token` might be needed